tests/conftest.py

```diff
@@ -29,6 +49,84 @@ def sapinvoices_client(sapinvoices_app):
     return sapinvoices_app.test_client()
 
 
+@pytest.fixture
+def ecs_client(mock_ecs_cluster, mock_ecs_task_definition, mock_ecs_network_config):
+    return ECSClient()
+
+
+@pytest.fixture
+def mock_boto3_ecs_client():
+    with patch("webapp.utils.aws.ecs.boto3.client") as mock_client:
+        yield mock_client.return_value
+
+
+@pytest.fixture
+def mock_ecs_cluster():
+    """See https://github.com/getmoto/moto/blob/master/tests/test_ecs/test_ecs_boto3.py."""
+    with mock_aws():
+        ecs = boto3.client("ecs", region_name=AWS_DEFAULT_REGION)
+        ecs.create_cluster(clusterName="mock-sapinvoices-ecs-test")
+        yield
+
+
+@pytest.fixture
+def mock_ecs_task_definition():
+    """See https://github.com/getmoto/moto/blob/master/tests/test_ecs/test_ecs_boto3.py."""
+    with mock_aws():
+        ecs = boto3.client("ecs", region_name=AWS_DEFAULT_REGION)
+
+        # Registering with minimal definition
+        definition = {
+            "family": "mock-sapinvoices-ecs-test",
+            "containerDefinitions": [
+                {
+                    "name": "mock-sapinvoices-ecs-test",
+                    "image": "mock-sapinvoices-test:latest",
+                    "memory": 400,
+                }
+            ],
+        }
+        ecs.register_task_definition(**definition)
+        yield
+
+
+@pytest.fixture
+def mock_ecs_network_config():
+    return {
+        "awsvpcConfiguration": {
+            "securityGroups": ["sg-abc123"],
+            "subnets": ["subnet-abc123"],
+        }
+    }
+
+
+@pytest.fixture
+def mock_ecs_task_state_transitions(
+    mock_ecs_cluster, mock_ecs_task_definition, mock_ecs_network_config
+):
+    """See https://github.com/getmoto/moto/blob/master/tests/test_ecs/test_ecs_boto3.py."""
+    state_manager.set_transition(
+        model_name="ecs::task", transition={"progression": "manual", "times": 1}
+    )
```

Comment:
This is really cool!  I wasn't aware of a state manager that can cycle through known states for an AWS asset.  I'm going to dig into this a bit more in my own time, become familiar with it, but nice find.  Seems like a perfect use of it.